openeo/rest/job.py

```diff
@@ -424,6 +418,43 @@ def load_bytes(self) -> bytes:
     # TODO: more `load` methods e.g.: load GTiff asset directly as numpy array
 
 
+    def _download_to_file(self, url: str, target: Path, chunk_size: int):
+        head = self.job.connection.head(url, stream=True)
+        if head.ok and 'Accept-Ranges' in head.headers and 'bytes' in head.headers['Accept-Ranges']:
+            file_size = int(head.headers['Content-Length'])
+            self._download_chunked(url=url, target=target, file_size=file_size, chunk_size=chunk_size)
+        else:
+            self._download_unchunked(url=url, target=target)
+
+
+    def _download_chunked(self, url: str, target: Path, file_size: int, chunk_size: int):
```

Comment:
I think some things are mixed up here. Originally we already had chunked downloading (`for block in response.iter_content`), but what is added in this feature is "range request" based downloading, so the naming of these private methods is a bit confusing.